### PR TITLE
Add circleci configs

### LINKED ISSUE #388: Remove Cirrus files and compile iOS app on CircleCI
The cirrus CI did not compile, therefore I removed the integration from this Github project.

We need to move the compilation to circleci and remove old files from cirrus.

### PR DESCRIPTION
**Pull request by maxammann**
fixes #388

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -31,9 +33,110 @@ workflows:
...
+          key: flutter-1-{{ checksum ".fvm/fvm_config.json" }}-{{ arch }}
+          paths:
+            - .fvm
+      - run:
+          name: Configure Flutter
+          command: |
+            fvm --verbose flutter config --no-analytics
+      - restore_cache:
+          keys:
+            - pub-1-{{ checksum "pubspec.lock" }}-{{ arch }}
```
Comment (maxammann) [inline .circleci/config.yml:73]: The caches are not doing much, but I'm not sure which files to cache

**Comment by maxammann** [comment]
fixes #388

File: .circleci/config.yml [hunk 2]
```diff
@@ -252,3 +359,6 @@ jobs:
             echo $APT_HOST_FINGERPRINT > known_hosts
             echo "Uploading: " /tmp/workspace/debs/*.deb
             sftp -b - -o UserKnownHostsFile=known_hosts ci@apt.ehrenamtskarte.app:/local-apt-repository/ \<<< "put -r /tmp/workspace/debs/*.deb"
+
+https://github:
+  com/m0nac0/flutter-maplibre-gl/pull/9:
```
Comment (klinzo) [inline .circleci/config.yml:364]:
```suggestion
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -31,9 +33,114 @@ workflows:
...
+            fvm flutter precache --ios
+      - save_cache:
+          key: pub-2-{{ checksum "pubspec.lock" }}-{{ arch }}
+          paths:
+            - .dart_tool
+      - run:
+          name: Update Pods
+          command: |
+            cd ios
+            pod update # https://github.com/m0nac0/flutter-maplibre-gl/pull/9
```
Comment (klinzo) [inline .circleci/config.yml:135]: What is this comment about?

File: .circleci/config.yml [hunk 4]
```diff
@@ -31,9 +33,114 @@ workflows:
...
+            pod install
+      - run:
+          name: Build
+          command: |
+            fvm --verbose flutter build ios --no-codesign --release -t lib/main_prod.dart
```
Comment (klinzo) [inline .circleci/config.yml:140]: Why don't you save the .ipa as well?

File: frontend/ios/Runner.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -892,6 +893,7 @@
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
 				ENABLE_NS_ASSERTIONS = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				"EXCLUDED_ARCHS[sdk=*]" = armv7;
```
Comment (klinzo) [inline frontend/ios/Runner.xcodeproj/project.pbxproj:896]: Is this correct?

**Review by klinzo — APPROVED** [review]
Is this still up to date:
> Make sure you have Flutter version `2.0.6` installed,
as we currently depend on Dart Sdk `<2.13.0`.

from https://github.com/digitalfabrik/ehrenamtskarte/blob/main/frontend/README.md
since https://github.com/digitalfabrik/ehrenamtskarte/blob/main/frontend/.fvm/fvm_config.json says to use flutter sdk 2.5.2

File: .circleci/config.yml (hunk 4 above)
Comment (maxammann) [reply .circleci/config.yml:140]: No IPA is build here, also just an ipa is quite useless if it is not signed with an ad-hoc certificate.

File: frontend/ios/Runner.xcodeproj/project.pbxproj (hunk 5 above)
Comment (klinzo) [reply frontend/ios/Runner.xcodeproj/project.pbxproj]: Some lines above/below there is also another change. Perhaps check the whole project.pbxproj for unwanted changes

Comment (maxammann) [reply frontend/ios/Runner.xcodeproj/project.pbxproj]: everything fine :)

**Pull request merged**